nexus/reconfigurator/execution/src/clickhouse.rs

```diff
@@ -38,7 +40,20 @@ use std::str::FromStr;
 
 const CLICKHOUSE_DATA_DIR: &str = "/data";
 
-pub(crate) async fn deploy_nodes<'a, I>(
+pub(crate) async fn deploy_nodes(
+    opctx: &OpContext,
+    blueprint: &Blueprint,
+    clickhouse_cluster_config: &ClickhouseClusterConfig,
+) -> Result<(), Vec<anyhow::Error>> {
+    deploy_nodes_impl(
+        opctx,
+        blueprint.all_omicron_zones(BlueprintZoneDisposition::any),
```

Comment:
I know you're just moving this - but should this disposition be `is_in_service`? Why is `any` acceptable here?

Reply:
> I know you're just moving this - but should this disposition be `is_in_service`? Why is `any` acceptable here?

John kept this in because the keeper code was doing it's own filtering, and the goal was not to change any behavior. But looking at this execution code again, it does seem wrong. There's no need to send updates to expunged zones. I was worried that the keeper config would be in correct, but this doesn't change that config, it only changes which keepers we send the config to.

Reply:
It's not obvious to me that this set of zones is only used for which sleds to talk to - maybe it is? E.g., it looks like the IPs of these zones end up sent to everyone - maybe that's also wrong if they're expunged, but I know there was a lot of nitpicky stuff about how much can be changed at once for clickhouse. Do you mind if I file an issue and let you or Karen take a look at changing this?

Reply:
Yes, please file an issue and I'll take care of it. Thanks!